topiary-queries/queries/openscad.scm

```diff
@@ -0,0 +1,385 @@
+    (assign_block)
+    (union_block)
+    (use_statement)
+    (include_statement)
+    (assert_statement)
+  ] @append_spaced_softline
+  .
+  [
+    "else"
+    (block_comment)
+    (line_comment)
+  ]* @do_nothing
+)
+
+(line_comment) @append_hardline
+
+(block_comment) @multi_line_indent_all
+
+; Allow line break after block comments
+(
+  (block_comment)
+  .
+  _ @prepend_input_softline
+)
+
+; Append softlines, unless followed by comments.
+; When binding multiple values in a let block, allow new lines between the bindings.
+(list
+  "[" @append_indent_start @append_empty_softline @append_antispace
+  "]" @prepend_indent_end @prepend_empty_softline @prepend_antispace
+)
+; to avoid having a list that directly follows a let_expression look visually unindented,
+; add another level of indentation:
+; let (
+;   u = true,
+; )[
+;  [1, 0, 0, 0],
+;  [0, 1, 0, 0],
+;  [0, 0, 1, 0],
+;  [0, 0, 0, 1],
+; ];
+(let_expression
+  (list
+    "[" @append_indent_start
+    "]" @prepend_indent_end
+  )
+)
+
+(range
+  "[" @append_antispace
+  "]" @prepend_antispace
+)
+(list "," @append_spaced_softline . [(block_comment) (line_comment)]* @do_nothing)
+(assignments "," @append_spaced_softline . [(block_comment) (line_comment)]* @do_nothing)
+(parameters "," @append_spaced_softline . [(block_comment) (line_comment)]* @do_nothing)
+(";" @append_spaced_softline . [(block_comment) (line_comment)]* @do_nothing)
+
+; Never put a space before a comma
+("," @prepend_antispace)
+(";" @prepend_antispace)
+
+; Don't insert spaces between the operator and their expression operand
+; '-x' v.s. '- x'
+(unary_expression _ @append_antispace . (expression))
+
+; ================================================================================
+; functions & modules
+; ================================================================================
+(function_item "function" @append_space)
+
+; indent the body of a function
+(function_item
+  (parameters)
+  .
+  "=" @append_spaced_softline @append_indent_start
+  (expression)
+  ";" @prepend_indent_end
+)
+
+; ===
+; function literals
+; ===
+(function_lit (parameters) @append_spaced_softline)
+(function_call (parenthesized_expression) @append_antispace (arguments))
+
+; ===
+; module_item, module_call, and transform_chain
+; ===
+
+; everything except `union_block` after a for/if/else statement should be a spaced_softline
+(module_item
+  body: [
+    (for_block)
+    (intersection_for_block)
+    (if_block)
+    (let_block)
+    (assign_block)
+    (transform_chain)
+    (include_statement)
+    (assert_statement)
+  ] @prepend_indent_start @append_indent_end
+)
+
+; module calls in a transformation chain will follow each other
+; sometimes staying on the same line and sometimes having a linebreak,
+; each linebreak typically also starts an indent scope
+(transform_chain) @prepend_input_softline
+(transform_chain
+  (modifier)*
+  (module_call) @append_indent_start
+  (transform_chain) @append_indent_end
+)
```

Comment:
My hunch is that the double-indent is happening because of one/both of these queries.

Reply:
yeah see https://github.com/tweag/topiary/pull/845#discussion_r1932656458 and the comment on line 190 & 191, this was explicit.